### PR TITLE
[#47] Add `doctest` tests to `co-log-core` package

### LINKED ISSUE #47: Add `doctest` tests to `co-log-core` package
Blocked by #46 

It's a great idea to test all functions in `Colog.Core` package using `doctest`. After #46 is implemented, `logStringStdout` action can be used as testing example. This will improve documentation significantly.

### PR DESCRIPTION
**Pull request by lucazulian**
Resolves #47

### CODE REVIEW

**Comment by ChShersh** [comment]
@lucazulian Thank you for the contribution! But there's already open PR #54 by @tfausak that adds `doctest` support to `co-log-core`. I think your contribution is still valuable because you've added more tests to the existing functions and I like your formatting of the `test-suite` stanza! So I can propose to wait until @tfausak's PR is merged (if he would like to continue his work) and then you can rebase on top of the new `master`.

**Comment by lucazulian** [comment]
Thank you @ChShersh,
right now I'm tring to figure out why build fails. It seems thaht there is a conflict with time dependency.
I'm tring with cabal new-build all --enable-tests --max-backjumps=-1 --reorder-goals command and this is what I get:
```
Resolving dependencies...                                                        
cabal: Could not resolve dependencies:                                           
[__0] trying: co-log-0.2.0 (user goal)                                           
[__1] trying: base-4.12.0.0/installed-4.1... (dependency of co-log)              
[__2] next goal: time (dependency of co-log)                                     
[__2] rejecting: time-1.8.0.2/installed-1.8... (conflict: co-log =>              
time^>=1.9.2)                                                                    
[__2] trying: time-1.9.2                                                         
[__3] trying: co-log-core-0.1.1 (user goal)                                      
[__4] rejecting: co-log-core:!test (constraint from config file, command line    
flag, or user target requires opposite flag selection)                           
[__4] trying: co-log-core:*test                                                  
[__5] trying: doctest-0.16.0.1 (dependency of co-log-core *test)                 
[__6] next goal: ghc (dependency of doctest)                                     
[__6] rejecting: ghc-8.6.1/installed-8.6... (conflict: time==1.9.2, ghc =>       
time==1.8.0.2/installed-1.8...)                                                  
[__6] rejecting: ghc-8.6.1 (conflict: time==1.9.2, ghc => time>=1.4 && <1.9)     
[__6] rejecting: ghc-8.4.3, ghc-8.4.1 (conflict:                                 
base==4.12.0.0/installed-4.1..., ghc => base<0 && ==4.11.*)                      
[__6] rejecting: ghc-8.2.2, ghc-8.2.1 (conflict:                                 
base==4.12.0.0/installed-4.1..., ghc => base<0 && ==4.10.*)                      
After searching the rest of the dependency tree exhaustively, these were the     
goals I've had most trouble fulfilling: ghc, doctest, base, co-log, time,        
co-log-core, co-log-core:test                                                    
```

**Comment by ChShersh** [comment]
@lucazulian Regarding failing CI: see my comment here: https://github.com/kowainik/co-log/pull/54#pullrequestreview-162249648

**Comment by ChShersh** [comment]
@lucazulian Okay, I think you can now rebase on `master` :slightly_smiling_face: Feel free to ask any questions about `git`as well!

**Comment by lucazulian** [comment]
@ChShersh thank you for your review.
Probably in the next few days I will add more `doctest` in the package.

File: co-log-core/co-log-core.cabal [hunk 1]
```diff
@@ -47,6 +47,6 @@ library
 test-suite doctest
   build-depends: base, doctest ^>= 0.16.0
   default-language: Haskell2010
-  hs-source-dirs: tests
-  main-is: doctest.hs
+  hs-source-dirs: test
+  main-is: Doctests.hs
   type: exitcode-stdio-1.0
```
Comment (ChShersh) [inline co-log-core/co-log-core.cabal]: Could you, please, format `test-suite` with alignment as it was previously in your PR? :slightly_smiling_face:

File: co-log-core/src/Colog/Core/IO.hs [hunk 2]
```diff
@@ -3,18 +3,29 @@ module Colog.Core.IO
     , logStringStderr
     , logStringHandle
     , withLogStringFile
+    , liftLogIO
```
Comment (ChShersh) [inline co-log-core/src/Colog/Core/IO.hs]: Thanks for exporting this function!

File: co-log-core/src/Colog/Core/IO.hs [hunk 3]
```diff
@@ -3,18 +3,29 @@ module Colog.Core.IO
...
 import System.IO (Handle, IOMode( AppendMode ), hPutStrLn, stderr, withFile)
 import Colog.Core.Action (LogAction (..))
 
 
-{- | Action that prints 'String' to stdout. -}
+{- | Action that prints 'String' to stdout.
+
+>>> unLogAction logStringStdout "foo"
+foo
+
```
Comment (ChShersh) [inline co-log-core/src/Colog/Core/IO.hs]: Actually, empty line before `-}` is not needed. So you can safely remove it, it's redundant. If you see some empty lines in other places in documentation before `-}` you can remove them as well :+1:

**Comment by ChShersh** [comment]
@lucazulian Thanks for your work! I left couple comments. Having doctests for functions from `Colog.Core.Actions` is extremely useful addition. I would like to see tests for every function in this module. This will be sooo cool :sparkles: If you would like to work on them, let me know!

**Comment by lucazulian** [comment]
@ChShersh I absolutely want to work on it, and also I'm trying to figure out what kind of tests to write for #22 so I do not overlap with the doctests. Thank you again for the support.

**Comment by ChShersh** [comment]
@lucazulian Oh, don't worry about #22 issue. Tests from #22 are for `co-log` package, while `doctest` is only for `co-log-core` so tests won't overlap anyway :slightly_smiling_face:

**Pull request merged**